lib/widgets/editor/editor_toolbar.dart

```diff
@@ -0,0 +1,462 @@
+              }),
+              icon: store.imageUploadState.isLoading
+                  ? const CircularProgressIndicator.adaptive()
+                  : const Icon(Icons.image),
+              tooltip: L10n.of(context).editor_image,
+            );
+          },
+        ),
+        IconButton(
+          onPressed: () async {
+            final person = await PickPersonDialog.show(context);
+
+            if (person != null) {
+              final name =
+                  '@${person.person.name}@${person.person.originInstanceHost}';
+              final link = person.person.actorId;
+
+              controller.reformatSimple('[$name]($link)');
+            }
+          },
+          icon: const Icon(Icons.person),
+          tooltip: L10n.of(context).editor_user,
+        ),
+        IconButton(
+          onPressed: () async {
+            final community = await PickCommunityDialog.show(context);
+            if (community != null) {
+              final name =
+                  '!${community.community.name}@${community.community.originInstanceHost}';
+              final link = community.community.actorId;
+
+              controller.reformatSimple('[$name]($link)');
+            }
+          },
+          icon: const Icon(Icons.home),
+          tooltip: L10n.of(context).editor_community,
+        ),
+        PopupMenuButton<HeaderLevel>(
+          itemBuilder: (context) => [
+            for (final h in HeaderLevel.values)
+              PopupMenuItem(
+                value: h,
+                child: Text(describeEnum(h).toUpperCase()),
+              ),
+          ],
+          onSelected: (val) {
+            final header = '${'#' * val.value} ';
+
+            if (!controller.firstSelectedLine.startsWith(header)) {
+              controller.insertAtBeginningOfFirstSelectedLine(header);
+            }
+          },
+          tooltip: L10n.of(context).editor_header,
+          child: const Icon(Icons.h_mobiledata),
+        ),
+        IconButton(
+          onPressed: () => controller.surround(
+            before: '~~',
+            placeholder: L10n.of(context).insert_text_here_placeholder,
+          ),
+          icon: const Icon(Icons.format_strikethrough),
+          tooltip: L10n.of(context).editor_strikethrough,
+        ),
+        IconButton(
+          onPressed: () {
+            controller.insertAtBeginningOfEverySelectedLine('> ');
+          },
+          icon: const Icon(Icons.format_quote),
+          tooltip: L10n.of(context).editor_quote,
+        ),
+        IconButton(
+          onPressed: () {
+            final line = controller.firstSelectedLine;
+
+            if (line.startsWith(RegExp.escape('* '))) {
+              controller.removeAtBeginningOfEverySelectedLine('* ');
+            } else if (line.startsWith('- ')) {
+              controller.removeAtBeginningOfEverySelectedLine('- ');
+            } else {
+              controller.insertAtBeginningOfEverySelectedLine('- ');
+            }
```

Comment:
`+`

It would be nice if you kept all of these markdown markers in a centralized place. Then this code should handle a list of symbols instead of hardcoding some ifs

Reply:
bump

Reply:
i'll see what i can do

Reply:
done